modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileFeatureSource.java

```diff
@@ -382,12 +370,8 @@ protected int getCountInternal(Query query) throws IOException {
             reader.setFilter(filter);
         }
 
-        // setup the target bbox if any, and the generalization hints if available
+        // setup the generalization hints if available
         if (q != null) {
-            if (bbox != null && !bbox.isNull()) {
```

Comment:
Hi @aaime, the assigment of the filter to reader in https://github.com/geotools/geotools/pull/1056/files#diff-06c0794a50eeb31f89a201b87d5ca19cR370 and subsequent validation in the reader does my code valid? In this case the filter and the targetbox are not redundant?